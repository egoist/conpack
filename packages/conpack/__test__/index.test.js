import Conpack from '../src'

it('simple', () => {
  const conpack = new Conpack()
  conpack.config.entry = 'src/index.js'
  const jsRule = conpack.rules.add('js', {
    test: /\.js$/,
    exclude: [/node_modules/]
  })
  jsRule.loaders.add('babel', {
    loader: 'babel-loader',
    options: {}
  })
  jsRule.loaders.add('eslint', {
    loader: 'eslint-loader',
    options: { eslintrc: false }
  })
  jsRule.loaders.replace('babel', 'buble', {
    loader: 'buble',
    options: { hehe: true }
  })
  jsRule.loaders.update('eslint', options => {
    options.loader = 'my-eslint-loader'
    return options
  })
  class Plugin {
    constructor(...options) {
      this.options = options
    }
  }
  conpack.plugins.add('uglify', Plugin, [{ foo: 1 }, 's'])
  expect(conpack.toConfig()).toMatchSnapshot()
})
