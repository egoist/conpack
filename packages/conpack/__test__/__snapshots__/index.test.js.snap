// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple 1`] = `
Object {
  "entry": "src/index.js",
  "module": Object {
    "rules": Array [
      Object {
        "exclude": Array [
          /node_modules/,
        ],
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "buble",
            "options": Object {
              "hehe": true,
            },
          },
          Object {
            "loader": "my-eslint-loader",
            "options": Object {
              "eslintrc": false,
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [
    Plugin {
      "options": Array [
        Object {
          "foo": 1,
        },
        "s",
      ],
    },
  ],
}
`;
